### ISSUE TITLE
Invalid Avro header does not raise error event

### ISSUE THREAD

**Issue by craxal** [mtth/avsc#448]
I discovered some mysterious behavior while using `BlockDecoder`. If the header is invalid (`!tap.isVaid()`), it does not raise an error. Instead, it just returns.

This was confusing, because I was expecting an error to be thrown so I could process that and provide a helpful message elsewhere.

**Comment by mtth** [comment]
Agreed that it's surprising behavior for the stream to not emit an error if it is `end`ed without a valid header. (Until that happens, the `tap.isValid` check is expected: it just means the header is missing data.)

Until this is fixed your workaround sounds right. To contain complexity on your end (and make migration easier later on) you could isolate the logic in a thin factory function. Something like (untested):

```js
function newBlockDecoder(...args) {
  let truncated = true;
  return new avro.streams.BlockDecoder(...args)
    .on('finish', function () { if (truncated) { this.emit('error', new Error('Invalid header')); }})
    .on('metadata', () => { truncated = false; });
}
```
